nikola/nikola.py

```diff
@@ -930,6 +929,23 @@ def init_plugins(self, commands_only=False, load_all=False):
                         bad_candidates.add(p)
                         utils.LOGGER.debug('Not loading compiler extension {}', p[-1].name)
             self.plugin_manager._candidates = list(set(self.plugin_manager._candidates) - bad_candidates)
+
+        # Find repeated plugins and discard the less local copy
+        def plugin_position_in_places(plugin):
+            for i, place in enumerate(self._plugin_places):
+                if plugin[0].startswith(place):
+                    return i
+
+        plugin_dict = defaultdict(list)
+        for data in self.plugin_manager._candidates:
+            plugin_dict[data[2].name].append(data)
```

Comment:
they are "path to the .plugin file, path to the module, metadata about the
plugin"

On Sat, Jun 4, 2016 at 1:37 PM Chris Warrick <notifications@github.com>
wrote:

> In nikola/nikola.py
> <https://github.com/getnikola/nikola/pull/2364#discussion_r65804619>:
>
> > @@ -930,6 +929,23 @@ def init_plugins(self, commands_only=False, load_all=False):
> >                          bad_candidates.add(p)
> >                          utils.LOGGER.debug('Not loading compiler extension {}', p[-1].name)
> >              self.plugin_manager._candidates = list(set(self.plugin_manager._candidates) - bad_candidates)
> > +
> > +        # Find repeated plugins and discard the less local copy
> > +        def plugin_position_in_places(plugin):
> > +            for i, place in enumerate(self._plugin_places)
>
> :
> > +                if plugin[0].startswith(place):
> > +                    return i
> > +
> > +        plugin_dict = defaultdict(list)
> > +        for data in self.plugin_manager._candidates:
> > +            plugin_dict[data[2].name].append(data)
>
> why [2], what are the previous positions?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/getnikola/nikola/pull/2364/files/10067d382d48c9c171fc6cece55dff9c203e917f#r65804619>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAGK8IiUkS8LYFmTiPZDyQOWCqeA4Vdks5qIanPgaJpZM4IuKiY>
> .
>